lib/pi_piper/bcm2835.rb

```diff
@@ -1,5 +1,15 @@
-require "pi_piper/bcm2835/version"
+require 'pi_piper/bcm2835/version'
+require 'pi_piper/bcm2835/pin'
+require 'pi_piper/bcm2835/spi'
+require 'pi_piper/bcm2835/i2c'
+require 'pi_piper/bcm2835/driver'
 
 module PiPiper
-  autoload PiPiper::Bcm2835, "pi_piper/bcm2835/bcm2835"
+  module Bcm2835
+    class << self
+      def driver
+        @driver ||= PiPiper::Bcm2835::Driver.new
+      end
```

Comment:
And as you only need one driver to be loaded my implementation allow to only write : 
```ruby
require 'pi_piper'
require 'pi_piper-bcm2835'
```

it load the code and load the driver with `PiPiper.driver = PiPiper::Bcm2835` in the Bcm2835 module/class